colorcoder/ColorPairImpl.java

```diff
@@ -0,0 +1,29 @@
+package colorcoder;
+
+public class ColorPairImpl {
+
+	public ColorPair getColorFromPairNumber(int pairNumber) {
+		int zeroBasedPairNumber = pairNumber - 1;// 3
+		MajorColor majorColor = MajorColor.fromIndex(zeroBasedPairNumber / MajorColor.values().length);
+		MinorColor minorColor = MinorColor.fromIndex(zeroBasedPairNumber % MinorColor.values().length);
+		return new ColorPair(majorColor, minorColor);
+	}
+
+	public int getPairNumberFromColor(MajorColor major, MinorColor minor) {
+		return major.getIndex() * MinorColor.values().length + minor.getIndex() + 1;
+	}
+
+	public void testNumberToPair(int pairNumber, MajorColor expectedMajor, MinorColor expectedMinor) {

```

Comment:
It is a better approach always to keep test code separate from production code. You may write a new class containing test code and consume colorPairImpl class for testing.